### PR TITLE
Add metrics.

### PR DESCRIPTION
**Pull request by jmjoy**
Add metrics support.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -167,7 +173,7 @@ For details, please refer to [prost-build:sourcing-protoc](https://docs.rs/prost
 
 # Release
 
-The SkyWalking committer(PMC included) could follow [this doc](Release-guide.md) to release an official version.
+The SkyWalking committer(PMC included) could follow [this doc](https://github.com/apache/skywalking-rust/blob/master/Release-guide.md) to release an official version.
```
Comment (jmjoy) [inline README.md:176]: Make it also accessable in crates.io and docs.rs.

File: README.md [hunk 2]
```diff
@@ -47,6 +47,12 @@ the context after the span finished.
 
 LogRecord is the simple builder for the LogData, which is the Log format of Skywalking.
 
+## Metrics
+
+### MeterRecord
+
+MeterRecord is the simple builder for the MeterData, which is the metrics format of Skywalking.
```
Comment (wu-sheng) [inline README.md:54]: You should provide a full list of the metric data types with explanations.

We have a short version on main repo docs, https://skywalking.apache.org/docs/main/latest/en/protocols/readme/#metrics which targets to proto file.

**Comment by wu-sheng** [comment]
From a metrics APIs pespective, to ease the usage, the single value metric is usually separated as counter or gauge to provide different use patterns.

You could refer Java meter APIs from here, https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/java-plugin-development-guide/#meter-plugin

File: README.md (hunk 2 above)
Comment (jmjoy) [reply README.md]: Updated.

**Comment by jmjoy** [comment]
I have refer to  java agent and update the API.

**Comment by wu-sheng** [comment]
Just confirm, all metrics objects(counter, gauge, histogram), are they thread safe? Such as concurrency write, and write and report concurrently.

**Comment by jmjoy** [comment]
> Just confirm, all metrics objects(counter, gauge, histogram), are they thread safe? Such as concurrency write, and write and report concurrently.

Yes, because they all implement `Transform`, which need `Send` and `Sync`, and the method `increment` of `Counter`,  method `add_value` of `Histogram`, only need `&self` (the inner use atomic).

**Review by wu-sheng — APPROVED** [review]
LGTM

**Pull request merged**